src/main/scala/mesosphere/marathon/MarathonSchedulerActor.scala

```diff
@@ -167,31 +167,6 @@ class MarathonSchedulerActor private (
     case cmd @ Deploy(plan, force) =>
       deploy(sender(), cmd)
 
-    case cmd @ KillTasks(runSpecId, tasks) =>
-      @SuppressWarnings(Array("all")) /* async/await */
-      def killTasks(): Future[Event] = {
-        logger.debug("Received kill tasks {} of run spec {}", tasks, runSpecId)
-        async {
-          await(killService.killInstances(tasks, KillReason.KillingTasksViaApi))
-          await(schedulerActions.scale(runSpecId))
```

Comment:
A scale check will be initiated for each task that is reportedly terminal so the `schedulerActions.scale(runSpecId)` call here was pretty redundant.